packages/starrating/src/react/index.js

```diff
@@ -117,8 +118,11 @@ class StarRating extends React.PureComponent {
   render() {
     const { starCount, value } = this.props
 
+    const filteredProps = filterReactProps(this.props)
+    delete filteredProps.onChange
```

Comment:
Is this `delete` pattern going to be oft-required?  I imagine that you're saying here "I use onChange to mean something else in another place in the app. I don't want the user to be able to attache it to this div".  

This seems like one of the cognitive challenges of letting "all" props in... we have to anticipate if *any* of those that the user passes in will cause a problem with our code.  

Seems like we'll have to do this often.  What's your take?

Reply:
hopefully it's not used to often but I don't know enough to say for sure. I get how this is a side effect of allowing all props...something to continue to consider...